### PR TITLE
Update ray-lightning with the latest PL Strategy API changes

### PR DESCRIPTION
**Pull request by kaushikb11**
Hey @amogkam @richardliaw!

With Lightning v1.5, we have introduced a new Trainer Strategy API. You could read more about it in this [blog](https://devblog.pytorchlightning.ai/announcing-the-new-lightning-trainer-strategy-api-f70ad5f9857e).

I have implemented the required updates in this PR. Do check it out and let me know if you have any questions! Thanks.

### CODE REVIEW

**Comment by richardliaw** [comment]
Hey @kaushikb11, seems like some of the examples still do `Trainer(plugin=RayStrategy)` -- should it be `Trainer(strategy=RayStrategy)`?

File: ray_lightning/ray_ddp_sharded.py [hunk 1]
```diff
@@ -8,27 +8,27 @@
 
 from ray.util import PublicAPI
 
-from ray_lightning import RayPlugin
+from ray_lightning import RayStrategy
 
 
 # C3 linearization of parent classes will do breadth first since both
-# RayPlugin and DDPSpawnShardedPlugin share a common parent of DDPSpawnPlugin
+# RayStrategy and DDPSpawnShardedPlugin share a common parent of DDPSpawnPlugin
```
Comment (amogkam) [inline ray_lightning/ray_ddp_sharded.py:15]:
```suggestion
# RayStrategy and DDPSpawnShardedStrategy share a common parent of DDPSpawnStrategy
```

**Comment by amogkam** [comment]
Hey @kaushikb11 thanks a bunch for making this PR!

Although the `strategy` argument was added in 1.5, it seems like the `strategies` package and `DDPStrategy`, etc. classes are only going to be introduced in 1.6.

Would it make sense to hold off on this PR until the PTL 1.6 release?

**Comment by MarkusSpanring** [comment]
@kaushikb11 @amogkam really looking forward to seeing this PR in the next `ray_lightning` release! 
Since PTL 1.6 is now out when will be the earliest to expect it?

**Comment by rokrokss** [comment]
@amogkam could you revisit this?

**Comment by amogkam** [comment]
Just re-triggered CI- let's see if they pass!

**Comment by amogkam** [comment]
@kaushikb11 looks like the tests are still failing with PTL 1.6. Do you mind taking a look?

**Comment by JiahaoYao** [comment]
Hi all, the pr is already merged in another pr https://github.com/ray-project/ray_lightning/pull/163. this pr is closed.

**Pull request closed without merging**